src/Power.cpp

```diff
@@ -226,19 +224,24 @@ class AnalogBatteryLevel : public HasBatteryLevel
     {
 
         uint32_t raw = 0;
+        uint8_t raw_c = 0; //raw reading counter
 
 #ifndef BAT_MEASURE_ADC_UNIT // ADC1
-#ifdef ADC_CTRL
+#ifdef ADC_CTRL //enable adc voltage divider when we need to read
         pinMode(ADC_CTRL, OUTPUT);
-        digitalWrite(ADC_CTRL, HIGH);
+        digitalWrite(ADC_CTRL, ADC_CTRL_ENABLED);
         delay(10);
 #endif
         for (int i = 0; i < BATTERY_SENSE_SAMPLES; i++) {
-            raw += adc1_get_raw(adc_channel);
+            int val_ = adc1_get_raw(adc_channel);
+            if(val_ >= 0){ //save only valid readings
```

Comment:
Check for valid readings, invalid should give -1 but could be usefull to use strictly greather than 0 instead of >=.